# Load env

In [2]:
import os
os.chdir('..')

In [3]:
%load_ext autoreload
%autoreload 2

In [4]:
import sys
import importlib
from reports_etl import *
from enrich_holdings import *
from fossil_classification import *
from holdings_analysis import *
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
pd.options.display.float_format = "{:,.2f}".format

In [5]:
last_updated()

.py file last modified: 2021-12-16 18:36:31.498425


# Bulk download reports
Using the http response recorded while searching for reports here:
https://employersinfocmp.cma.gov.il/#/publicreports

In [20]:
# get reports from response.json search results
response_path = "data/downloaded reports/company reports 2021Q4/"
reports = get_reports_from_response(response_path)

Number of reports included in response.json: 59


## Check for missing reports

In [6]:
reports.head()

EntityId  CodeFile  DocumentCode  \
0       220        73      71100073   
1       219        73      71100073   
2      5937        73      71100073   
3       209        73      71100073   
4       293        73      71100073   

                                     DocumentName  DocumentId fileExt  \
0  רשימת נכסים ברמת נכס בודד- Public - חברה - גמל     2181007    xlsx   
1  רשימת נכסים ברמת נכס בודד- Public - חברה - גמל     2184207    xlsx   
2  רשימת נכסים ברמת נכס בודד- Public - חברה - גמל     2184890    xlsx   
3  רשימת נכסים ברמת נכס בודד- Public - חברה - גמל     2185960    xlsx   
4  רשימת נכסים ברמת נכס בודד- Public - חברה - גמל     2186050    xlsx   

   ReportFor  ReportPeriodCode ReportPeriodDesc  EntHistLevel  SystemCode  \
0          1          71000001     2021 רבעון 4             0      300003   
1          1          71000001     2021 רבעון 4             0      300003   
2          1          71000001     2021 רבעון 4             0      300003   
3          1          71000001     2021 רבעון 4             0      300003   
4          1          71000001     2021 רבעון 4             0      300003   

   FunctionCode  LicenseCode  LicenseStatusId  DateStart  DateFinish  \
0             0            0                0        nan         nan   
1             0            0                0        nan         nan   
2             0            0                0        nan         nan   
3             0            0                0        nan         nan   
4             0            0                0        nan         nan   

   ProductNum    LegalId  Name  ShortName   EventDate  StatusDate SystemName  \
0           0  570009449   nan        nan  01/10/2021  24/01/2022        גמל   
1           0  520034968   nan        nan  01/10/2021  30/01/2022        גמל   
2           0  514767490   nan        nan  01/10/2021  30/01/2022        גמל   
3           0  510694821   nan        nan  01/10/2021  31/01/2022        גמל   
4           0  520024985   nan        nan  01/10/2021  31/01/2022        גמל   

   ReportHistLevel  EventId  DocStatus  EntityName  LicenseId  \
0                0   624999          1         nan          0   
1                0   626177          1         nan          0   
2                0   626571          1         nan          0   
3                0   627222          1         nan          0   
4                0   627255          1         nan          0   

                                      ParentCorpName  InvestmentNumber  \
0        עוצ"מ - אגודה שיתופית לניהול קופות גמל בע"מ               nan   
1  החברה המנהלת של קרן השתלמות של עובדי חברת החשמ...               nan   
2                                    סלייס גמל בע"מ                nan   
3                                    לאומי קמ"פ בע"מ               nan   
4     עגור חברה לניהול קופות גמל וקרנות השתלמות בע"מ               nan   

   ParentCorpLegalId  ParentCorpEntlId  IsMaxRowsExceeded  \
0          570009449               nan              False   
1          520034968               nan              False   
2          514767490               nan              False   
3          510694821               nan              False   
4          520024985               nan              False   

                                                 url  
0  https://employersinfocmp.cma.gov.il/api/Public...  
1  https://employersinfocmp.cma.gov.il/api/Public...  
2  https://employersinfocmp.cma.gov.il/api/Public...  
3  https://employersinfocmp.cma.gov.il/api/Public...  
4  https://employersinfocmp.cma.gov.il/api/Public...

In [7]:
# filter specific reports for quick analysis
# reports = reports[reports["ParentCorpName"].str.startswith("מגדל")]

In [8]:
# download reports - run only once per response
download_reports(reports, response_path)

# Process multiple reports -> unified file

In [9]:
reports_fn_list = get_filename_list(response_path)

number of files to be pre-processed: 59


## Pre-process - check sheet names and column names, add to standardization if needed

In [10]:
pre_process_reports(reports_fn_list)

{'סכום נכסי הקרן': 59, 'מזומנים': 59, 'תעודות התחייבות ממשלתיות': 59, 'תעודות חוב מסחריות': 59, 'אג"ח קונצרני': 59, 'מניות': 59, 'קרנות סל': 59, 'קרנות נאמנות': 59, 'כתבי אופציה': 59, 'אופציות': 59, 'חוזים עתידיים': 59, 'מוצרים מובנים': 59, 'לא סחיר - תעודות התחייבות ממשלתי': 59, 'לא סחיר - תעודות חוב מסחריות': 59, 'לא סחיר - אג"ח קונצרני': 59, 'לא סחיר - מניות': 59, 'לא סחיר - קרנות השקעה': 59, 'לא סחיר - כתבי אופציה': 59, 'לא סחיר - אופציות': 59, 'לא סחיר - חוזים עתידיים': 59, 'לא סחיר - מוצרים מובנים': 59, 'הלוואות': 59, 'פקדונות מעל 3 חודשים': 59, 'זכויות מקרקעין': 59, 'השקעה בחברות מוחזקות': 59, 'השקעות אחרות': 59, 'יתרת התחייבות להשקעה': 59, 'עלות מתואמת אג"ח קונצרני סחיר': 59, 'עלות מתואמת אג"ח קונצרני ל.סחיר': 59, 'עלות מתואמת מסגרות אשראי ללווים': 59, '{PL}PickLst': 3}


מזומנים  תעודות התחייבות ממשלתיות  \
שם המנפיק/שם נייר ערך       59.00                     59.00   
מספר ני"ע                   59.00                     59.00   
מספר מנפיק                  59.00                       nan   
דירוג                       59.00                     59.00   
שם מדרג                     59.00                     59.00   
סוג מטבע                    59.00                     59.00   
שעור ריבית                  59.00                     59.00   
תשואה לפדיון                59.00                     59.00   
שווי                        59.00                     59.00   
שעור מנכסי אפיק ההשקעה      59.00                     59.00   
שעור מסך נכסי השקעה         59.00                     59.00   
זירת מסחר                     nan                     59.00   
תאריך רכישה                   nan                     59.00   
מח"מ                          nan                     59.00   
ערך נקוב                      nan                     59.00   
שער                           nan                     59.00   
פדיון/ריבית/דיבידנד לקבל      nan                     59.00   
שעור מערך נקוב מונפק          nan                     59.00   
ספק מידע                      nan                       nan   
ענף מסחר                      nan                       nan   
ASX                           nan                       nan   
אוסטרליה                      nan                       nan   
בנקים וחברות אחזקה            nan                       nan   
אירו                          nan                       nan   
נכס בסיס                      nan                       nan   
קונסורציום כן/לא              nan                       nan   
מעלות                         nan                       nan   
שקל חדש                       nan                       nan   
תאריך שערוך אחרון             nan                       nan   
אופי הנכס                     nan                       nan   
שעור תשואה במהלך התקופה       nan                       nan   
כתובת הנכס                    nan                       nan   
ריבית אפקטיבית                nan                       nan   
Energy                        nan                       nan   

                          תעודות חוב מסחריות  אג"ח קונצרני  מניות  קרנות סל  \
שם המנפיק/שם נייר ערך                  59.00         59.00  59.00     59.00   
מספר ני"ע                              59.00         59.00  59.00     59.00   
מספר מנפיק                             59.00         59.00  59.00     59.00   
דירוג                                  59.00         59.00    nan       nan   
שם מדרג                                59.00         59.00    nan       nan   
סוג מטבע                               59.00         59.00  59.00     59.00   
שעור ריבית                             59.00         59.00    nan       nan   
תשואה לפדיון                           59.00         59.00    nan       nan   
שווי                                   59.00         59.00  59.00     59.00   
שעור מנכסי אפיק ההשקעה                 59.00         59.00  59.00     59.00   
שעור מסך נכסי השקעה                    59.00         59.00  59.00     59.00   
זירת מסחר                              59.00         59.00  59.00     59.00   
תאריך רכישה                            59.00         59.00    nan       nan   
מח"מ                                   59.00         59.00    nan       nan   
ערך נקוב                               59.00         59.00  59.00     59.00   
שער                                    59.00         59.00  59.00     59.00   
פדיון/ריבית/דיבידנד לקבל               55.00         59.00  59.00     59.00   
שעור מערך נקוב מונפק                   59.00         59.00  59.00     59.00   
ספק מידע                               59.00         59.00  59.00       nan   
ענף מסחר                               59.00         59.00  59.00     59.00   
ASX                                      nan           nan    nan       nan   
אוסטרליה                                 nan           nan    nan       nan   
בנק

# process holding reports
1. summary sheets (for verification)
2. holding sheets - raw data

## 1. summary sheets

In [11]:
all_summary_sheets = process_summary_sheets(reports_fn_list)

In [12]:
totals = get_totals(all_summary_sheets)

Number of totals found: 59


In [13]:
totals["sum_num"].sum()

1776366424.8264937

## 2. Holdings

In [14]:
# extract all holdings from reports
all_holdings_clean = clean_holdings(extract_holdings(reports_fn_list))

Processing report 59 out of 59
before cleaning: 88752
 after cleaning: 80031


# add fund and company data to holdings

In [15]:
## Manually add missing corps to reports - only if needed, for funds processing
missing_corps = pd.DataFrame(columns=reports.columns)
# missing_corps = missing_corps.append({'SystemName': 'חיים ואובדן כושר עבודה',
#                                       'ParentCorpName': 'מנורה מבטחים ביטוח בע"מ',
#                                       'ParentCorpLegalId': '520042540',
#                                       'ReportPeriodDesc': ''
#                                      }, ignore_index=True)
# missing_corps = missing_corps.append({'SystemName': 'חיים ואובדן כושר עבודה',
#                                       'ParentCorpName': 'כלל חברה לביטוח בע"מ',
#                                       'ParentCorpLegalId': '520024647',
#                                       'ReportPeriodDesc': ''
#                                      }, ignore_index=True)

reports = reports.append(missing_corps)

In [16]:
all_holdings_final = add_report_data(all_holdings_clean, reports)

# Compare totals from summary sheets with totals from holdings

In [17]:
print("total from summary sheets: {:,.2f}".format(totals["sum_num"].sum()))
print("total from holdings sheets: {:,.2f}".format(all_holdings_final["שווי"].sum()))
print("diff: {:,.2f}".format(totals["sum_num"].sum() - all_holdings_final["שווי"].sum()))

total from summary sheets: 1,776,366,424.83
total from holdings sheets: 1,769,113,338.76
diff: 7,253,086.06


In [18]:
totals_from_holdings = pd.DataFrame(all_holdings_final.groupby("report_id")["שווי"].agg('sum')).reset_index()
totals_from_holdings.index = totals_from_holdings.index.astype('str')

In [19]:
totals_comparison = totals_from_holdings.merge(totals,
                                               on='report_id',
                                               how='outer'
                                              )

totals_comparison["diff"] = totals_comparison["sum_num"] - totals_comparison["שווי"]
totals_comparison[abs(totals_comparison["diff"]) > 1].sort_values("diff", ascending=False)

report_id           שווי                          asset            sum pct  \
45   2222108  42,550,199.32  סה"כ סכום נכסי המסלול או הקרן  49,723,257.06   1   
55   2225626 213,128,906.45  סה"כ סכום נכסי המסלול או הקרן 213,189,352.44   1   
11   2216901   2,442,477.37  סה"כ סכום נכסי המסלול או הקרן   2,459,446.12   1   
41   2221683  80,583,303.57  סה"כ סכום נכסי המסלול או הקרן  80,584,342.30   1   
58   2226019  93,772,952.27  סה"כ סכום נכסי המסלול או הקרן  93,773,952.90   1   
56   2225937  48,704,983.60  סה"כ סכום נכסי המסלול או הקרן  48,705,555.83   1   

   pct_num        sum_num         diff  
45       1  49,723,257.06 7,173,057.74  
55       1 213,189,352.44    60,445.98  
11       1   2,459,446.12    16,968.75  
41       1  80,584,342.30     1,038.72  
58       1  93,773,952.90     1,000.63  
56       1  48,705,555.83       572.23

In [20]:
# report drill down
def report_sums_by_type(report_id):
    rep = all_holdings_final[all_holdings_final["report_id"] == report_id]
    print("total sum by holdings: {:,.2f}".format(rep["שווי"].sum()))
    return pd.DataFrame(rep.groupby("holding_type")["שווי"].agg('sum').map('{:,.2f}'.format))
    
report_sums_by_type('2222108')

total sum by holdings: 42,550,199.32


שווי
holding_type                                   
אג"ח קונצרני                       2,557,722.98
אופציות                               95,780.07
הלוואות                            1,996,889.98
השקעות אחרות                          -1,353.65
זכויות מקרקעין                       246,315.01
חוזים עתידיים                         27,735.64
כתבי אופציה                           46,987.83
לא סחיר - אג"ח קונצרני               313,755.14
לא סחיר - אופציות                     27,259.27
לא סחיר - חוזים עתידיים              190,595.35
לא סחיר - כתבי אופציה                    685.16
לא סחיר - מוצרים מובנים              469,465.52
לא סחיר - מניות                      601,386.31
לא סחיר - קרנות השקעה              2,966,480.28
לא סחיר - תעודות התחייבות ממשלתי  13,859,593.13
לא סחיר - תעודות חוב מסחריות          14,944.33
מוצרים מובנים                         30,875.91
מזומנים                              254,274.97
מניות                             12,629,952.22
עלות מתואמת אג"ח קונצרני סחיר          8,990.48
פקדונות מעל 3 חודשים                 347,326.34
קרנות נאמנות                       1,298,829.48
קרנות סל                           3,587,049.97
תעודות התחייבות ממשלתיות             976,394.73
תעודות חוב מסחריות                     2,262.87

### totals discrepancies
<u>2021Q2</u><br>
* In report 2128339 - there's a redundant line in נדלן, consider removing manually when all files for Q3 are here - removing
* In report 2153994 - the holdings in מזומנים are mostly missing. adding an artificial row. (הפניקס)
* In report 2146293 - the value of holdings in לא סחיר - אג"ח קונצרני are way off. leaving as is for now.

<u>2021Q3</u>
* 2148642 - missing exactly 10000 in מזומנים. leaving as is.
* 2148994 - dropped a line in הלוואות withno security number. not fixing for now.

<u>2021Q3</u>
* 2222108 - the holdings in מזומנים are mostly missing. adding an artificial row. (הפניקס)


## Removing redundant rows

In [20]:
remove_rows = (
    (all_holdings_final["report_id"] == '2128339') &
    (all_holdings_final["holding_type"] == 'זכויות מקרקעין') &
    (all_holdings_final["שעור מנכסי אפיק ההשקעה"] == 1)
)

all_holdings_final = all_holdings_final.drop(all_holdings_final[remove_rows].index, axis=0)

# all_holdings_final.groupby("report_id").filter(
#     lambda x: x["שעור מנכסי אפיק ההשקעה"].sum() > 1
# ).head()

## Adding missing rows

In [20]:
# adding a row for report 2153994, to compensate for missing holding values
total_from_summary = all_summary_sheets[
    (all_summary_sheets["report_id"] == '2222108') &
    (all_summary_sheets["asset"]).str.contains('מזומנים')
]["sum_num"]

holdings_to_fix = all_holdings_final[
    (all_holdings_final["report_id"] == '2222108') &
    (all_holdings_final["holding_type"] == 'מזומנים')
]

total_from_holdings = holdings_to_fix["שווי"].sum()

# calculate sum and pct to be added
to_be_added_sum = float(total_from_summary) - float(total_from_holdings)
to_be_added_pct = 1 - holdings_to_fix["שעור מנכסי אפיק ההשקעה"].sum()
to_be_added_pct_of_report = float(to_be_added_sum) / float(totals[totals["report_id"] == '2222108']["sum_num"])

# prepare new row - take the highest amount row as base and edit it
new_row = holdings_to_fix.sort_values("שווי", ascending=False).head(1)
new_row["שם המנפיק/שם נייר ערך"] = "unknown - added to match sum"
new_row['שווי'] = to_be_added_sum
new_row['שעור מנכסי אפיק ההשקעה'] = to_be_added_pct
new_row['שעור מסך נכסי השקעה'] = to_be_added_pct_of_report

for col in ['מספר ני"ע', 'מספר מנפיק', 'דירוג', 'שם מדרג', 'סוג מטבע']:
    new_row[col] = None

print("adding row: ")
new_row

adding row: 


שם המנפיק/שם נייר ערך מספר ני"ע מספר מנפיק דירוג שם מדרג  \
63490  unknown - added to match sum      None       None  None    None   

      סוג מטבע שעור ריבית תשואה לפדיון         שווי  שעור מנכסי אפיק ההשקעה  \
63490     None          0            0 7,172,886.70                    0.97   

       שעור מסך נכסי השקעה report_id holding_type זירת מסחר תאריך רכישה מח"מ  \
63490                 0.14   2222108      מזומנים       NaN         NaN  NaN   

      ערך נקוב  שער פדיון/ריבית/דיבידנד לקבל שעור מערך נקוב מונפק ספק מידע  \
63490      NaN  NaN                      NaN                  NaN      NaN   

      ענף מסחר נכס בסיס קונסורציום כן/לא תאריך שערוך אחרון אופי הנכס  \
63490      NaN      NaN              NaN               NaN       NaN   

      שעור תשואה במהלך התקופה כתובת הנכס ריבית אפקטיבית SystemName  \
63490                     NaN        NaN            NaN      פנסיה   

                ParentCorpName ParentCorpLegalId ProductNum Name ShortName  \
63490  הפניקס פנסיה וגמל בע"מ          513026484          0  NaN       NaN   

       StatusDate ReportPeriodDesc  
63490  31/03/2022     2021 רבעון 4

In [21]:
# adding the artificial row
all_holdings_final = all_holdings_final.append(new_row)

In [22]:
all_holdings_final.to_csv(response_path+"all_holdings.csv", index=False)

In [23]:
all_holdings_ever = all_holdings_final

## Export file for classification

In [24]:
cols_for_cls = [
    'שם המנפיק/שם נייר ערך', 'מספר ני"ע', 'מספר מנפיק',
    'שווי' ,'שעור מנכסי אפיק ההשקעה', 'שעור מסך נכסי השקעה', 
       'holding_type', 'זירת מסחר', 'תאריך רכישה', 'ערך נקוב', 'שער','שעור מערך נקוב מונפק',
       'ענף מסחר',
       'SystemName', 'ParentCorpName', 'ReportPeriodDesc'
]

holdings_for_cls = all_holdings_final[
    all_holdings_final["holding_type"].isin(['מניות', 'אג"ח קונצרני'])
]

holdings_for_cls[cols_for_cls].to_csv(response_path+"holdings_for_cls.csv", index=False)

In [29]:
holdings_for_cls[
    (holdings_for_cls['מספר ני"ע'].astype(str) == '11616780')
]

שם המנפיק/שם נייר ערך מספר ני"ע מספר מנפיק דירוג    שם מדרג סוג מטבע  \
56194  אלדן תחבורה אגח ו-הפחתה  11616780  510454333   ilA  S&P מעלות  שקל חדש   
65367  אלדן תחבורה אגח ו-הפחתה  11616780  510454333   ilA  S&P מעלות  שקל חדש   

      שעור ריבית תשואה לפדיון     שווי שעור מנכסי אפיק ההשקעה  \
56194       0.03         0.00   324.90                   0.00   
65367       0.03         0.00 3,562.14                   0.00   

      שעור מסך נכסי השקעה report_id  holding_type זירת מסחר תאריך רכישה מח"מ  \
56194                   0   2222108  אג"ח קונצרני      TASE    30/04/21 2.88   
65367                   0   2221683  אג"ח קונצרני      TASE    30/04/21 2.88   

          ערך נקוב    שער פדיון/ריבית/דיבידנד לקבל שעור מערך נקוב מונפק  \
56194   324,058.60 100.26                        0                 0.00   
65367 3,552,903.03 100.26                        0                 0.01   

      ספק מידע ענף מסחר נכס בסיס קונסורציום כן/לא תאריך שערוך אחרון אופי הנכס  \
56194      NaN  שירותים      NaN              NaN               NaN       NaN   
65367      NaN  שירותים      NaN              NaN               NaN       NaN   

      שעור תשואה במהלך התקופה כתובת הנכס ריבית אפקטיבית SystemName  \
56194                     NaN        NaN            NaN      פנסיה   
65367                     NaN        NaN            NaN      ביטוח   

                ParentCorpName ParentCorpLegalId ProductNum Name ShortName  \
56194  הפניקס פנסיה וגמל בע"מ          513026484          0  NaN       NaN   
65367  הפניקס חברה לביטוח בע"מ         520023185          0  NaN       NaN   

       StatusDate ReportPeriodDesc  
56194  31/03/2022     2021 רבעון 4  
65367  31/03/2022     2021 רבעון 4

## add new processed holdings to all_holdings file

In [10]:
# reload results from file
all_holdings_path = fetch_all_holdings_path()
new_holdings_path = "data/downloaded reports/company reports 2021Q4/all_holdings.csv"
all_holdings_ever = concat_from_csv_by_path(all_holdings_path, new_holdings_path)

pd.DataFrame(all_holdings_ever.groupby(['ReportPeriodDesc'])['שווי'].agg('sum').map('{:,.2f}'.format))

/Users/urimarom/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2850: DtypeWarning: Columns (6,7) have mixed types.Specify dtype option on import or set low_memory=False.
  if self.run_code(code, result):
/Users/urimarom/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2850: DtypeWarning: Columns (7) have mixed types.Specify dtype option on import or set low_memory=False.
  if self.run_code(code, result):


שווי
ReportPeriodDesc                  
2020 רבעון 1      1,222,418,614.88
2020 רבעון 2      1,307,193,119.07
2020 רבעון 3      1,373,544,876.41
2020 רבעון 4      1,461,378,236.82
2021 רבעון 1      1,548,168,300.50
2021 רבעון 2      1,593,940,074.16
2021 רבעון 3      1,640,889,505.45
2021 רבעון 4      1,776,286,225.46

In [11]:
# update all holdings file - handle with care
# TODO: add backup process
all_holdings_ever.to_csv(fetch_all_holdings_path(), index=False)

# Validate holdings file

In [34]:
all_holdings_curr_q = all_holdings_ever[all_holdings_ever["ReportPeriodDesc"] == '2021 רבעון 4']
# pd.DataFrame(all_holdings_curr_q.groupby(['SystemName', 'ParentCorpName', 'report_id'])['שווי'].agg('sum').map('{:,.2f}'.format))
pd.DataFrame(all_holdings_curr_q.groupby(['ParentCorpName'])['שווי'].agg('sum').map('{:,.2f}'.format))

שווי
ParentCorpName                                                    
איי. די. איי. חברה לביטוח בע"מ                          320,449.63
ארם גמולים - חברה לניהול קופות גמל בע''מ              1,237,284.76
גל -ניהול קופות גמל לעובדי הוראה בע"מ                 4,768,951.39
החברה המנהלת של מינהל קרן ההשתלמות לפקידים עובד...      931,420.12
החברה המנהלת של קרן השתלמות של עובדי חברת החשמל...      727,422.89
החברה המנהלת של רום קרן ההשתלמות לעובדי הרשויות...    4,274,459.64
החברה לניהול קופת התגמולים והפנסיה של עובדי הסו...    5,294,090.65
החברה לניהול קרן ההשתלמות להנדסאים וטכנאים בע"מ       2,144,250.78
החברה לניהול קרן השתלמות לביוכימאים  ומקרוביולו...      314,045.61
הכשרה חברה לביטוח בע"מ                               24,761,410.60
הלמן-אלדובי חח"י גמל בע"מ                            45,376,097.76
הנדסאים וטכנאים - חברה לניהול קופות גמל בע"מ          1,115,648.75
הפניקס חברה לביטוח בע"מ                              80,583,303.57
הפניקס פנסיה וגמל בע"מ                              112,554,747.55
הראל חברה לביטוח בע"מ                                75,477,332.01
הראל פנסיה וגמל בע"מ                                171,458,126.75
חברה לניהול קופות גמל של העובדים בעיריית תל - א...      159,615.60
יהב - פ.ר.ח. - חברה לניהול קופות גמל בע"מ               179,307.16
יהב אחים ואחיות - חברה לניהול קופות גמל בע"מ          2,280,300.42
יהב רופאים - חברה לניהול קופות גמל בע"מ               1,340,843.32
ילין לפידות ניהול קופות גמל בע"מ                     47,479,624.93
לאומי קמ"פ בע"מ                                         190,463.41
לעתיד חברה לניהול קרנות פנסיה בע"מ                    1,469,270.90
מגדל חברה לביטוח בע"מ                               143,536,792.19
מגדל מקפת קרנות פנסיה וקופות גמל בע"מ               123,591,749.91
מור גמל ופנסיה בע"מ                                  29,818,516.93
מחוג - מינהל גמל לעובדי חברת חשמל לישראל בע"מ         2,759,171.13
מחר - חברה לניהול קופות גמל בע"מ                      1,380,212.93
מיטב דש גמל ופנסיה בע"מ                              78,265,614.50
מנורה מבטחים ביטוח בע"מ                              35,481,682.27
מנורה מבטחים והסתדרות המהנדסים ניהול קופות גמל ...    2,442,477.37
מנורה מבטחים פנסיה וגמל בע"מ                        218,328,275.51
סלייס גמל בע"מ                                          779,158.83
עגור חברה לניהול קופות גמל וקרנות השתלמות בע"מ        7,329,112.54
עו"ס - חברה לניהול קופות גמל בע"מ                       227,934.10
עוצ"מ - אגודה שיתופית לניהול קופות גמל בע"מ             657,407.20
ק.ל.ע. - חברה לניהול קרן השתלמות לעובדים סוציאל...      723,754.32
קו הבריאות חברה לניהול קופות גמל בע"מ                 1,764,957.48
קופת תגמולים של עובדי אל על נתיבי אוויר לישראל ...      894,586.88
קופת תגמולים של עובדי התעשיה האוירית לישראל בע"מ      2,021,058.65
קופת"ג של עובדי עירית חיפה                              263,032.12
קרן החסכון לצבא הקבע - חברה לניהול קופות גמל בע"מ     6,332,582.42
קרנות השתלמות למורים ולגננות - חברה מנהלת בע"מ       21,975,100.12
קרנות השתלמות למורים תיכוניים, מורי סמינרים ומפ...    7,901,962.33
רעות חברה לניהול קופות גמל בע"מ                       2,367,627.99
שיבולת - חברה לניהול קופות גמל בע"מ                   1,177,930.21
תגמולים של עובדים בעירית ת"א-יפו א.ש. בע"מ            1,146,197.68

## Look for missing company reports

In [25]:
reports_cnt = pd.DataFrame(all_holdings_ever.groupby(['SystemName', 'ParentCorpName', 'ParentCorpLegalId'])['ReportPeriodDesc'].nunique().reset_index())
period_cnt = all_holdings_ever["ReportPeriodDesc"].nunique()

companies_with_missing_reports = reports_cnt[reports_cnt["ReportPeriodDesc"] < period_cnt]

In [26]:
available_reports_periods = pd.DataFrame(all_holdings_ever.groupby(['SystemName', 'ParentCorpName', 'ParentCorpLegalId', 'ReportPeriodDesc'])['שווי'].sum()).reset_index()
companies_w_missing_reports_available_reports = pd.merge(
    companies_with_missing_reports[['ParentCorpLegalId', 'SystemName']],
    available_reports_periods,
    on = ['ParentCorpLegalId', 'SystemName'],
    how="inner"
)
companies_w_missing_reports_available_reports

Empty DataFrame
Columns: [SystemName, ParentCorpName, ParentCorpLegalId, ReportPeriodDesc, שווי]
Index: []

Located missing reports for major companies, manually downloaded

# Add fossil classification
Based on quarterly company level reports classification

## get previous fossil classifications

In [12]:
prev_cls_fn = "data_sources/prev_class.csv"
prev_by_sec_num, prev_by_ISIN = get_latest_fossil_classifications(prev_cls_fn)

previously classified Israeli security nums by is_fossil:
0.00    2306
1.00     232
nan        1
Name: is_fossil, dtype: int64
previously classified ISINs by is_fossil:
0.00    6653
1.00     555
Name: is_fossil, dtype: int64


## directly using previous company reports
Could be useful for fund level reports, commented out for now

In [13]:
# # TODO: using different file per holding_type for now, should change
# q_company_report1 = pd.read_csv('/Users/urimarom/Downloads/חשיפה לפוסיליים - Q2 2021 - החזקות ישירות באגח קונצרני.csv')
# q_company_report2 = pd.read_csv('/Users/urimarom/Downloads/חשיפה לפוסיליים - Q2 2021 - החזקות ישירות במניות.csv')

# q_company_report = pd.concat([
#     q_company_report1[['מספר ני"ע', 'is_fossil']],
#     q_company_report2[['מספר ני"ע', 'is_fossil']]
# ])

# q_company_report = q_company_report.drop_duplicates(subset=['מספר ני"ע']).set_index('מספר ני"ע')
# q_company_report.index = q_company_report.index.astype('str')

## Using previous classification

In [22]:
all_holdings_ever = pd.read_csv(fetch_all_holdings_path(), dtype=str)

In [24]:
all_holdings_cls = add_fossil_classifications(all_holdings_ever, prev_by_sec_num, prev_by_ISIN)

all_holdings: 666242
having holding number: 563510
without holding number: 102732


/Users/urimarom/PycharmProjects/fossil_classification/reports_etl.py:537: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  holdings_with_num['מספר ני"ע'] = holdings_with_num['מספר ני"ע'].astype('str').str.strip().str.upper()


Holdings after fossil classification by Israeli security num:
nan     387167
0.00    154283
1.00     22060
Name: is_fossil, dtype: int64
Holdings after fossil classification by ISIN:
nan     304398
0.00    231154
1.00     27958
Name: is_fossil, dtype: int64
total fossil sum: 424941867.95378697
holdings count before classification: 666242
holdings count after classification: 666242


## Validate Fossil Classification results

In [29]:
# look for fossil holdings that aren't מניות or אג"ח קונצרני
all_holdings_cls[
    (all_holdings_cls["is_fossil"] > 0) &
    (~all_holdings_cls["holding_type"].isin(["מניות", 'אג"ח קונצרני']))
]

# looking good!
# TODO: might want to try and catch additional fossil holdings of other types - join by all id types

שם המנפיק/שם נייר ערך     מספר ני"ע  \
915                                     רותם אנרגיה אפ 3       1155746   
1349                                שיכון ובינוי6  מתואם       1129733   
3098                                      SPDR S&P OIL &  US78468R5569   
4066                      ORGS US - אופ לא סחירה 20.1.23     707705711   
10559                                    רותם אנרגיה אפ3       1155746   
11494  XOP US SPDR S&P Oil & Gas Expl- SPDR S&P OIL &...  US78468R5569   
15747                                       אלומיי  אפ 1       1169325   
19672                                 שפיר הנדסה אג"ח ב'       1141951   
19681                                        דור אלון ה'       1136761   
20787  XOP US SPDR S&P Oil & Gas Expl- SPDR S&P OIL &...  US78468R5569   
22402                                    רותם אנרגיה אפ3       1155746   
25026                                 שפיר הנדסה אג"ח ב'       1141951   
25035                                        דור אלון ה'       1136761   
25383                                 שיכון ובינוי נעמ 2       1183052   
27040  Israel electric 4% 19/06/28- חברת החשמל לישראל...  XS0085848421   
30309                                     SPDR S&P O&G E  US78468R5569   
31076                       רותם אנרגיה אופ 3- Rotem OPC       1155746   
38414                                 שפיר הנדסה אג"ח ב'       1141951   
38425                                        דור אלון ה'       1136761   
38506                         CANADIAN NATURAL RESOURCES  CA1363851017   
38913                                    רותם אנרגיה אפ3       1155746   
40078                                       אלומיי אופ 1       1169325   
45846                                 שיכון ובינוי נעמ 2       1183052   
46985                                    רותם אנרגיה אפ3       1155746   
48715                    אלומיי  אפ 1- אלומיי קפיטל בע"מ       1169325   
48828                            LVIATH 5 3/4 30/06/2023  IL0011677668   
48829                             ICLIT 6.375 31/05/2038  IL0028103310   
58182                                       אלומיי אופ 1       1169325   
59882                                 Israel electric 4%  XS0085848421   
63281                                    רותם אנרגיה אפ3       1155746   
65321                               GL LN C 370 18/03/22  JE00B4T3BW64   
65322                               GL LN C 440 18/03/22  JE00B4T3BW64   
65323                               GL LN P 320 18/03/22  JE00B4T3BW64   
70395                                 שפיר הנדסה אג"ח ב'       1141951   
70404                                        דור אלון ה'       1136761   
78981                                    אלומיי אופציה 1       1169325   
79060                               GL LN C 370 18/03/22  JE00B4T3BW64   
79061                               GL LN C 440 18/03/22  JE00B4T3BW64   
79062                               GL LN P 320 18/03/22  JE00B4T3BW64   
81942  XOP US SPDR S&P Oil & Gas Expl- SPDR S&P OIL &...  US78468R5569   
88182                                    שיכוןבינוי נעמ2       1183052   
88607                                       אלומיי אופ 1       1169325   

      מספר מנפיק  דירוג    שם מדרג      סוג מטבע שעור ריבית תשואה לפדיון  \
915          NaN    NaN        NaN       שקל חדש        NaN          NaN   
1349         NaN    ilA  S&P מעלות       שקל חדש       0.04          NaN   
3098         NaN    NaN        NaN  דולר אמריקאי        NaN          NaN   
4066         NaN    NaN        NaN  דולר אמריקאי        NaN          NaN   
10559        NaN    NaN        NaN       שקל חדש        NaN          NaN   
11494               NaN        NaN  דולר אמריקאי        NaN          NaN   
15747        NaN    NaN        NaN       שקל חדש        NaN          NaN   
19672        NaN   ilA+  S&P מעלות       שקל חדש       0.03            0   
19681        NaN  A3.il     מידרוג       שקל חדש       0.05            0   
20787        NaN    NaN        NaN  דולר אמריקאי        NaN          NaN   
22402        NaN

In [18]:
should_be_cls = all_holdings_cls[all_holdings_cls["holding_type"].isin(["מניות", 'אג"ח קונצרני'])]
missing_fossil_cls = pd.DataFrame(
    should_be_cls["is_fossil"].isnull().groupby([should_be_cls["ParentCorpName"], should_be_cls["ReportPeriodDesc"]]).sum()
)
missing_fossil_cls

is_fossil
ParentCorpName                                     ReportPeriodDesc           
איי. די. איי. חברה לביטוח בע"מ                     2020 רבעון 1              0
                                                   2020 רבעון 2              0
                                                   2020 רבעון 3              0
                                                   2020 רבעון 4              0
                                                   2021 רבעון 1              0
                                                   2021 רבעון 2              0
                                                   2021 רבעון 3              0
                                                   2021 רבעון 4              0
איילון חברה לביטוח בע"מ                            2020 רבעון 1              0
                                                   2020 רבעון 2              0
                                                   2020 רבעון 3              0
                                                   2020 רבעון 4              0
                                                   2021 רבעון 1              0
                                                   2021 רבעון 2              0
                                                   2021 רבעון 3              0
                                                   2021 רבעון 4              0
אינפיניטי ניהול השתלמות וגמל בע"מ                  2020 רבעון 1              0
                                                   2020 רבעון 2              0
                                                   2020 רבעון 3              0
                                                   2020 רבעון 4              0
                                                   2021 רבעון 1              0
                                                   2021 רבעון 2              0
                                                   2021 רבעון 3              0
אלטשולר שחם גמל ופנסיה בע"מ                        2020 רבעון 1              0
                                                   2020 רבעון 2              0
                                                   2020 רבעון 3              0
                                                   2020 רבעון 4              0
                                                   2021 רבעון 1              0
                                                   2021 רבעון 2              0
                                                   2021 רבעון 3              0
                                                   2021 רבעון 4              0
אנליסט קופות גמל בע"מ                              2020 רבעון 1              0
                                                   2020 רבעון 2              0
                                                   2020 רבעון 3              0
                                                   2020 רבעון 4              0
                                                   2021 רבעון 1              0
                                                   2021 רבעון 2              0
                                                   2021 רבעון 3              0
                                                   2021 רבעון 4              0
ארם גמולים - חברה לניהול קופות גמל בע''מ           2020 רבעון 1              0
                                                   2020 רבעון 2              0
                                                   2020 רבעון 3              0
                                                   2020 רבעון 4              0
                                                   2021 רבעון 1              0
                                                   2021 רבעון 2              0
                                                   2021 רבעון 3              0
                                                   2021 רבעון 4              0
גל -ניהול קופות גמל לעובדי הוראה בע"מ              2020 רבעון 1              0
                                                   2020 רבעון 2              0
                                        

## Write classified results to CSV & create another file with added id types

In [ ]:
all_holdings_cls.to_csv(response_path+"all_holdings_cls.csv", index=False)
# update all_holdings_cls_all_ids if needed
all_holdings_cls_all_ids = load_mappings_and_add_ids_to_holdings(all_holdings_cls)
all_holdings_cls_all_ids.to_csv(response_path+"all_holdings_cls_all_ids.csv", index=False)

/Users/urimarom/anaconda3/lib/python3.6/site-packages/pandas/core/strings.py:2001: UserWarning: This pattern has match groups. To actually get the groups, use str.extract.
  return func(self, *args, **kwargs)


## Get Fund stats

In [50]:
fund_stats = get_summary(all_holdings_cls, 'ParentCorpName', 'SystemName', 'ProductNum','ShortName')
fund_stats.sort_values('שווי פוסילי', ascending=False).head()

ParentCorpName SystemName ProductNum  \
831           מנורה מבטחים פנסיה וגמל בע"מ      פנסיה       2009   
531                  מגדל חברה לביטוח בע"מ      ביטוח      17012   
585  מגדל מקפת קרנות פנסיה וקופות גמל בע"מ      פנסיה       2102   
447                   כלל חברה לביטוח בע"מ      ביטוח      14012   
503                    כלל פנסיה וגמל בע"מ      פנסיה       2002   

                    ShortName          שווי  שווי פוסילי  \
831  מנורה מבטחים פנסיה כללי  97,517,828.84 3,688,669.73   
531                       NaN 75,632,465.16 2,936,796.68   
585           מקפת אישית כללי 63,232,444.53 1,807,664.74   
447                       NaN 44,269,431.59 1,725,016.47   
503      מיטבית עתודות - ספיר 53,073,440.42 1,608,629.20   

     שיעור פוסילי מסך הנכסים  שווי במניות ואגח קונצרני סחירים  \
831                     0.04                    29,373,735.39   
531                     0.04                    27,579,070.32   
585                     0.03                    18,360,816.57   
447                     0.04                    17,900,611.43   
503                     0.03                    17,753,944.08   

     שווי פוסילי במניות ואגח קונצרני סחירים  \
831                            3,688,669.73   
531                            2,936,796.68   
585                            1,807,664.74   
447                            1,725,016.47   
503                            1,608,629.20   

     שיעור פוסילי במניות ואגח קונצרני סחירים  שווי בסוגי החזקות לא פוסיליים  \
831                                     0.13                  36,290,878.50   
531                                     0.11                  11,209,943.92   
585                                     0.10                  22,850,094.41   
447                                     0.10                   8,481,832.16   
503                                     0.09                  17,847,738.78   

     שיעור פוסילי מתוך מניות ואגח סחירים + סוגי החזקות לא פוסיליים  
831                                               0.06              
531                                               0.08              
585                                               0.04              
447                                               0.07              
503                                               0.05

In [51]:
fund_stats.to_csv(response_path+"fund_stats.csv", index=False)

In [373]:
# get fossil holdings per fund

fund_stats[fund_stats["ProductNum"].duplicated()]
# TODO: there is 1 duplicate, fund number 103 - check it later
def get_fossil_holdings(fund_id):
    fossil_holdings = all_holdings_cls[
        (all_holdings_cls["ProductNum"] == fund_id) &
        (all_holdings_cls["is_fossil"] > 0)
    ]
    return fossil_holdings.sort_values("שווי פוסילי", ascending=False)

get_fossil_holdings(11366)

שם המנפיק/שם נייר ערך     מספר ני"ע מספר מנפיק   דירוג  \
528808                    איי.סי.אל        281014  520027830     NaN   
528800                 שיכון ובינוי       1081942  520036104     NaN   
528827                       פז נפט       1100007  510216054     NaN   
528839                  אלקו החזקות        694034  520025370     NaN   
528881                     דור אלון       1093202  520043878     NaN   
528661           שיכון ובינוי אגח 8       1135888  520036104     ilA   
528608                  חשמל אגח 27       6000210  520000472  Aa1.il   
528749             רציו מימון אגח ב       1139443  515060044       0   
528826                          בזן       2590248  520036658     NaN   
528607              חשמל     אגח 29       6000236  520000472  Aa1.il   
528795              או פי סי אנרגיה       1141571  514401702     NaN   
528755            דלק קידוחים אגח א       4750089  550013098   A2.il   
528753               תמר פטרו אגח ב       1143593  515334662   A1.il   
528720        אלון רבוע כחול אגח ד'       1139583  520042847   A3.il   
528840                       אקויטל        755017  520030859     NaN   
528845                     רציו יהש        394015  550012777     NaN   
528762             רציו מימון אגח ג       1142488  515060044       0   
528880                    ארקו קורפ       1170901    3535148     NaN   
528756          חברה לישראל אג"ח 11       5760244  520028010     ilA   
528663              או פי סי אגח ב'       1166057  514401702    ilA-   
528772           EDF 4 1/2 09/21/28  USF2893TAU00      27129    BBB+   
528660           שיכון ובינוי אגח 6       1129733  520036104     ilA   
528888                 תמר פטרוליום       1141357  515334662     NaN   
528754           תמר פטרוליום אגח א       1141332  515334662   A1.il   
528739                דלק קבוצה  לג       1138882  520044322     ilB   
528787         TRPCN 5 5/8 05/20/75  US89356BAA61      27376    Baa3   
528646         ג'נריישן קפיטל אגח ב       1177526  515846558    ilA+   
528715           חברה לישראל אגח 15       5760327  520028010     ilA   
528844                  ישראמקו יהש        232017  550010003     NaN   
528768          BPLN 3.543 04/06/27  US10373QBK58      28177      A-   
528645                 פז נפט אגח ז       1142595  510216054   ilAA-   
528644                    פז נפט  ו       1139542  510216054   ilAA-   
528719           שיכון ובינוי אגח 7       1129741  520036104     ilA   
528724              דור אלון  אגח ו       1140656  520043878   A3.il   
528776          Wplau 4.5% 04/03/29  USQ98229AN94      13112    BBB+   
528748             ג'נריישן קפ אגחא       1166222  515846558       0   
528677              אייסיאל   אגח ז       2810372  520027830    ilAA   
528738                דלק קב אגח לד       1143361  520044322     ilB   
528723                    בזן אגח י       2590511  520036658    ilA-   
528722                    בזן אגח ה       2590388  520036658    ilA-   
528714               חברה לישראל 10       5760236  520028010     ilA   
528750              ישראמקו נגב 2 א       2320174  550010003    ilAA   
528674  חברת חשמל 26 4.8% 2016/2023       6000202  520000472  Aa1.il   
528841                  חברה לישראל        576017  520028010     NaN   
528611               נתיבי גז אגח ד       1147503  513436394   ilAA+   
528757                    בזן אגח ו       2590396  520036658    ilA-   
528737              דלק קב   אגח לא       1134790  520044322     ilB   
528792         PEMEX 6 1/2 03/13/27  US71654QCG55      10333     Ba2   
528725               דור אלון אגח ה       1136761  520043878   A3.il   
528673                דלק קב אגח יט       1121326  520044322     ilB   
528915               DK US_Delek us  US24665A1034         54     NaN   
528843               דלק קדוחים יהש        475020  550013098     NaN   

          שם מדרג      סוג מטבע שעור ריבית תשואה לפדיון   שווי  \
528808        NaN       שקל חדש        NaN          NaN 931.63   
528800        NaN       שקל חדש        NaN          NaN 526.39   

## חסכון לכל ילד

In [116]:
fund_stats[
    fund_stats["ShortName"].str.contains("ילד", na=False)
].sort_values("שיעור פוסילי במניות ואגח קונצרני סחירים", ascending=False).reset_index()

index                         ParentCorpName SystemName  ProductNum  \
0     617                     מור קופות גמל בע"מ        גמל        9303   
1      87                  אנליסט קופות גמל בע"מ        גמל       11366   
2     799           מנורה מבטחים פנסיה וגמל בע"מ        גמל       11321   
3      86                  אנליסט קופות גמל בע"מ        גמל       11365   
4     255                הפניקס פנסיה וגמל בע"מ         גמל       11310   
5     800           מנורה מבטחים פנסיה וגמל בע"מ        גמל       11322   
6     927            פסגות קופות גמל ופנסיה בע"מ        גמל       11380   
7     182    הלמן - אלדובי קופות גמל ופנסיה בע"מ        גמל       11354   
8     183    הלמן - אלדובי קופות גמל ופנסיה בע"מ        גמל       11355   
9     568  מגדל מקפת קרנות פנסיה וקופות גמל בע"מ        גמל        9896   
10    257                הפניקס פנסיה וגמל בע"מ         גמל       11312   
11    184    הלמן - אלדובי קופות גמל ופנסיה בע"מ        גמל       11356   
12    354                   הראל פנסיה וגמל בע"מ        גמל       11375   
13    569  מגדל מקפת קרנות פנסיה וקופות גמל בע"מ        גמל        9897   
14    256                הפניקס פנסיה וגמל בע"מ         גמל       11311   
15    801           מנורה מבטחים פנסיה וגמל בע"מ        גמל       11323   
16    355                   הראל פנסיה וגמל בע"מ        גמל       11376   
17    356                   הראל פנסיה וגמל בע"מ        גמל       11377   
18    692                מיטב דש גמל ופנסיה בע"מ        גמל       11385   
19    693                מיטב דש גמל ופנסיה בע"מ        גמל       11386   
20    694                מיטב דש גמל ופנסיה בע"מ        גמל       11387   
21    570  מגדל מקפת קרנות פנסיה וקופות גמל בע"מ        גמל        9898   
22     39            אלטשולר שחם גמל ופנסיה בע"מ        גמל       11325   
23    928            פסגות קופות גמל ופנסיה בע"מ        גמל       11381   
24     88                  אנליסט קופות גמל בע"מ        גמל       11367   
25     40            אלטשולר שחם גמל ופנסיה בע"מ        גמל       11326   
26    929            פסגות קופות גמל ופנסיה בע"מ        גמל       11382   
27     41            אלטשולר שחם גמל ופנסיה בע"מ        גמל       11327   
28    180    הלמן - אלדובי קופות גמל ופנסיה בע"מ        גמל       11352   
29     42            אלטשולר שחם גמל ופנסיה בע"מ        גמל       11328   
30    616                     מור קופות גמל בע"מ        גמל        9113   
31    258                הפניקס פנסיה וגמל בע"מ         גמל       11335   
32    181    הלמן - אלדובי קופות גמל ופנסיה בע"מ        גמל       11353   
33    350                   הראל פנסיה וגמל בע"מ        גמל        9555   
34    567  מגדל מקפת קרנות פנסיה וקופות גמל בע"מ        גמל        9895   
35    618                     מור קופות גמל בע"מ        גמל        9414   
36    619                     מור קופות גמל בע"מ        גמל        9420   
37    620                     מור קופות גמל בע"מ        גמל        9421   
38    695                מיטב דש גמל ופנסיה בע"מ        גמל       11388   
39    798           מנורה מבטחים פנסיה וגמל בע"מ        גמל       11320   
40    846                        סלייס גמל בע"מ         גמל       11343   
41    847                        סלייס גמל בע"מ         גמל       11344   
42    848                        סלייס גמל בע"מ         גמל       11345   
43    930            פסגות קופות גמל ופנסיה בע"מ        גמל       11383   

                                            ShortName         שווי  \
0                             מור חיסכון לילד - שריעה    12,057.49   
1   אנליסט חיסכון לילד - חוסכים המעדיפים סיכון בינוני   101,480.54   
2   מנורה מבטחים חסכון לילד - חוסכים המעדיפים סיכו...   345,064.19   
3     אנליסט חיסכון לילד - חוסכים המעדיפים סיכון מועט   300,371.10   
4     אקסלנס חיסכון לילד - חוסכים המעדיפים סיכון מועט   305,991.72   
5   מנורה מבטחים חסכון לילד - חוסכים המעדיפים סיכו...   101,138.09   
6      פסגות חיסכון לילד - חוסכים המעדיפים סיכון מועט   347,506.79   
7   הלמן-אלדובי חיסכון לילד- חוסכים המעדיפים סיכון...   295,838.76  

In [411]:
all_holdings_cls.groupby(["SystemName","ParentCorpName","ProductNum"])["שעור מסך נכסי השקעה"].sum()

SystemName              ParentCorpName                         ProductNum
גמל                     אלטשולר שחם גמל ופנסיה בע"מ            472          1.00
                                                               558          1.00
                                                               588          1.00
                                                               1093         1.00
                                                               1094         1.00
                                                               1290         1.00
                                                               1375         1.00
                                                               1376         1.00
                                                               1377         1.00
                                                               1378         1.00
                                                               1394         1.00
                                   

## Fund level data - Discrepancies
* הלמן - seems like we looked at the wrong company level report - it was חברת חשמל...
<br>The numbers ~fit the ones in the company level PDF report. The gov site is missing a company level report.

the company name is: הלמן-אלדובי קופות גמל ופנסיה בע"מ


In [78]:
# missing reports
[r for r in funds_df.index.unique() if r not in all_holdings_final["report_id"].unique()]

# checked 2 missing, both have 0 holdings, all good

NameError: name 'funds_df' is not defined

In [28]:
pd.DataFrame(all_holdings_final.groupby(['ParentCorpName', 'SystemName'])["report_id"].nunique())

report_id
ParentCorpName                                     SystemName                       
איי. די. איי. חברה לביטוח בע"מ                     חיים ואובדן כושר עבודה          3
איילון חברה לביטוח בע"מ                            חיים ואובדן כושר עבודה         12
אלטשולר שחם גמל ופנסיה בע"מ                        גמל                            30
                                                   פנסיה                          14
אנליסט קופות גמל בע"מ                              גמל                            31
ארם גמולים - חברה לניהול קופות גמל בע''מ           גמל                             3
גל -ניהול קופות גמל לעובדי הוראה בע"מ              גמל                             9
החברה המנהלת של מינהל קרן ההשתלמות לפקידים עובד... גמל                             3
החברה המנהלת של קרן השתלמות של עובדי חברת החשמל... גמל                             1
החברה המנהלת של רום קרן ההשתלמות לעובדי הרשויות... גמל                             4
החברה לניהול קרן ההשתלמות להנדסאים וטכנאים בע"מ    גמל                             3
החברה לניהול קרן ההשתלמות לעובדי המדינה בע"מ       גמל                             5
החברה לניהול קרן השתלמות לאקדמאים במדעי החברה ו... גמל                             3
החברה לניהול קרן השתלמות לביוכימאים  ומקרוביולו... גמל                             2
החברה לניהול קרן השתלמות למשפטנים בע"מ             גמל                             3
החברה לניהול קרן השתלמות לשופטים בע"מ              גמל                             1
הכשרה חברה לביטוח בע"מ                             חיים ואובדן כושר עבודה         21
הלמן - אלדובי קופות גמל ופנסיה בע"מ                גמל                            39
הנדסאים וטכנאים - חברה לניהול קופות גמל בע"מ       גמל                             3
הפניקס חברה לביטוח בע"מ                            ביטוח                           2
                                                   חיים ואובדן כושר עבודה         28
הפניקס פנסיה וגמל בע"מ                             גמל                            46
                                                   פנסיה                          25
הראל חברה לביטוח בע"מ                              ביטוח                           4
                                                   חיים ואובדן כושר עבודה         30
הראל פנסיה וגמל בע"מ                               גמל                            43
                                                   פנסיה                          43
חברה לניהול קופות גמל של העובדים בעיריית תל - א... גמל                             3
חברת הגמל לעובדי האוניברסיטה העברית בע"מ           גמל                             5
יהב - פ.ר.ח. - חברה לניהול קופות גמל בע"מ          גמל                             2
יהב אחים ואחיות - חברה לניהול קופות גמל בע"מ       גמל                             2
יהב רופאים - חברה לניהול קופות גמל בע"מ            גמל                             3
ילין לפידות ניהול קופות גמל בע"מ                   גמל                            20
כלל חברה לביטוח בע"מ                               ביטוח                           2
                                                   חיים ואובדן כושר עבודה         19
כלל פנסיה וגמל בע"מ                                גמל                            36
                                                   פנסיה                          24
לאומי קמ"פ בע"מ                                    גמל                             1
מבטחים מוסד לביטוח סוציאלי של העובדים בע"מ         גמל                             1
מגדל חברה לביטוח בע"מ                              ביטוח                           4
                                                   חיים ואובדן כושר עבודה         15
מגדל מקפת קרנות פנסיה וקופות גמל בע"מ              גמל                            38
                                                   פנסיה                          27
מור קופות גמל בע"מ                                 גמל                            36
מחוג - מינהל גמל לעובדי חברת חשמל לישראל בע"מ      גמל                             4
מחר - חברה לניהול קופות גמל בע"מ                   גמל                          

### discrepancies

* report 2101262: empty sheet "זכויות מקרקעין" - explains all the diff
* report 2101276: all holding_types sums are ok, the sum @summary sheet is slightly different for some reason

In [123]:
print(funds_df[funds_df["ProductNum"] == 2207])
totals[totals["report_id"] == '2101262']

        SystemName                         ParentCorpName ParentCorpLegalId  \
2101262      פנסיה  מגדל מקפת קרנות פנסיה וקופות גמל בע"מ         512237744   

        ProductNum                                 Name  \
2101262       2207  מגדל מקפת אישית  למקבלי קצבה קיימים   

                             ShortName  StatusDate ReportPeriodDesc  
2101262  מקפת אישית למקבלי קצבה קיימים  29/08/2021     2021 רבעון 2  


asset          sum pct report_id  pct_num  \
4065  סה"כ סכום נכסי המסלול או הקרן  3.58272e+06   1   2101262      1.0   

           sum_num  
4065  3.582724e+06

# looking for lost insurance funds

### Migdal

In [37]:
migdal_ins = all_holdings_final[(all_holdings_final["ParentCorpLegalId"] == '520004896')]
migdal_by_fund = pd.DataFrame(migdal_ins.groupby(["ShortName", "ProductNum", "report_id"], dropna=False)["שווי"].agg('sum'))
migdal_by_fund.reset_index()

ShortName ProductNum              report_id  \
0                   מגדל - כללי       17013                2149427   
1                     מגדל אג"ח        8659                2149420   
2          מגדל אג"ח משולב מניות         74                2149430   
3         מגדל בסיסי למקבלי קצבה       9606                2149424   
4          מגדל מחקה מדד S&P 500      13625                2149423   
5    מגדל מסלול אג"ח ממשלת ישראל         68                2149417   
6   מגדל מסלול אג"ח עד 10% מניות         69                2149419   
7                מגדל מסלול הלכה        185                2149418   
8        מגדל מסלול לבני 50 ומטה       9599                2149422   
9       מגדל מסלול לבני 50 עד 60       9604                2149433   
10      מגדל מסלול לבני 60 ומעלה       9729                2149431   
11              מגדל מסלול מניות         76                2149416   
12           מגדל שיקלי טווח קצר       1955                2149421   
13                   מגדל-כללי 3         75                2149429   
14                    מסלול חו"ל       8660                2149432   
15                           NaN      17010  520004896_b17010_p321   
16                           NaN      17011  520004896_b17011_p321   
17                           NaN      17012  520004896_b17012_p321   
18                           NaN      17013  520004896_b17013_p321   

            שווי  
0  35,047,120.77  
1      93,094.92  
2   1,982,709.57  
3   5,828,863.43  
4      74,927.03  
5     678,881.44  
6   3,208,931.08  
7     649,674.93  
8   4,303,822.99  
9   2,059,060.55  
10  2,294,714.33  
11  2,893,698.61  
12    138,105.32  
13    499,217.75  
14     34,084.31  
15     30,055.28  
16  2,791,852.13  
17 75,632,465.16  
18 35,047,120.77

https://www.migdal.co.il/about/investments/assets - there are 4 more funds not included in the reports site:
* מגדל משתתף ברווחים - קרן ח - b010_p221_520004896
* מגדל משתתף ברווחים - קרן ט - 520004896_b011_p221
* מגדל משתתף ברווחים - קרן י - b012_p221_520004896
* מגדל משתתף קרן י החדשה - b013_p221_520004896

Manually downloaded, added to the directory.

In [ ]:
pd.DataFrame(all_holdings_final[
    (all_holdings_final["ParentCorpLegalId"].isnull()) |
    (all_holdings_final["ParentCorpLegalId"] == 520004896)
].groupby("report_id")['שווי'].agg('sum').map('{:,.2f}'.format))

### Menora missing reports

In [118]:
pd.DataFrame(all_holdings_final[all_holdings_final["ParentCorpName"] == 'מנורה מבטחים ביטוח בע"מ'
].groupby(["ShortName", "ProductNum"])['שווי'].agg('sum').map('{:,.2f}'.format)).sort_values("ProductNum").reset_index()

ShortName  ProductNum          שווי
0                   מנורה ביטוח שיקלי טווח קצר          41    110,717.59
1                    מנורה ביטוח אג"ח צמוד מדד          42  1,210,425.63
2                             מנורה ביטוח מט"ח          43     29,976.13
3                           מנורה ביטוח מניות           44    378,152.12
4                      מנורה ביטוח מניות בחו"ל         101    289,031.61
5                 מנורה ביטוח אג"ח ממשלת ישראל         180    306,301.40
6                       מנורה ביטוח מסלול הלכה         184    102,763.92
7               מנורה ביטוח בסיסי למקבלי קצבה         9534    738,535.53
8   מנורה ביטוח אג"ח עד 25% מניות למקבלי קצבה         9535  1,157,826.22
9                     מנורה ביטוח לבני 50 ומטה        9536  1,543,147.49
10                      מנורה ביטוח לבני 50-60        9537  1,061,029.11
11                  מנורה ביטוח לבני 60 ומעלה         9544  1,338,540.24
12                           מנורה ביטוח 20/80       13029     62,795.78

* total 17 funds in website, 2 of which mergerd into others --> 15.
<br><br>
<u>missing reports:</u>
1. 18011
1. 18012

download from here: https://www.menoramivt.co.il/rep-list/about-insurance/participating-portfolio-info/insurer-assets

In [ ]:
# adding sums of 2 migrated funds (data for these 2 is 1 quarter behind)
99576.34 + 531016.24 + all_holdings_final[all_holdings_final["ParentCorpName"] == 'מנורה מבטחים ביטוח בע"מ']["שווי"].sum()

Conclusion: there seems to be missing reports for Menora. Check again next quarter.

### הפניקס ביטוח - דוחות חסרים

In [127]:
pd.DataFrame(
    all_holdings_final[all_holdings_final["ParentCorpLegalId"] == '520023185'].groupby(["report_id", "ShortName", "ProductNum"])['שווי'].agg('sum').map('{:,.2f}'.format)
).sort_values("ProductNum", ascending=False).reset_index()

report_id                                   ShortName  ProductNum  \
0    2154066                     הפניקס ביטוח מניות חו"ל       12882   
1    2154068                     הפניקס ביטוח אג"ח בינ"ל       12881   
2    2154067                הפניקס - מסלול פאסיבי כללי\n       12092   
3    2154070       הפניקס - מסלול פאסיבי לבני 60 ומעלה\n       12091   
4    2154069       הפניקס - מסלול פאסיבי לבני 50 עד 60\n       12090   
5    2154071          הפניקס - מסלול פאסיבי לבני 50 ומטה       12089   
6    2154061        הפניקס-אקסלנס אינווסט שיקלי טווח קצר        9603   
7    2154059            הפניקס-אקסלנס אינווסט מסלול הלכה        9602   
8    2154072          הפניקס-אקסלנס אינווסט מסלול מניות         9601   
9    2154063            הפניקס-אקסלנס אינווסט מסלול כללי        9600   
10   2154065                  הפניקס מסלול לבני 60 ומעלה        9594   
11   2154064                  הפניקס מסלול לבני 50 עד 60        9593   
12   2154062                 הפניקס מסלול שיקלי טווח קצר        9592   
13   2154060                הפניקסמסלול הלכה למקבלי קצבה        9591   
14   2154058              הפניקס מסלול מניות למקבלי קצבה        9590   
15   2154056  הפניקס מסלול אג"ח עד 25% מניות למקבלי קצבה        9589   
16   2154052               הפניקס מסלול אג"ח למקבלי קצבה        9588   
17   2154050              הפניקס מסלול בסיסי למקבלי קצבה        9587   
18   2154048                           הפניקס מסלול הלכה        9586   
19   2154053              הפניקס מסלול אג"ח עד 25% מניות        9585   
20   2154054                          הפניקס מסלול מניות        9579   
21   2154057                 הפניקס מסלול מדדי אג"ח חו"ל        9556   
22   2154055                       הפניקס מחקה מדד sp500        9554   
23   2154046                 הפניקס - מסלול לבני 50 ומטה         471   
24   2154044     הפניקס-אקסלנס אינווסט אג"ח עד 25% מניות         104   
25   2154043      הפניקס-אקסלנס אינווסט תיק מנוהל אג"ח 2         103   
26   2154047                     הפניקס מסלול השקעה כללי          50   
27   2154045                           הפניקס-מסלול אג"ח          47   

             שווי  
0       93,198.05  
1      592,631.93  
2      397,250.12  
3      619,501.07  
4    1,136,081.56  
5    4,274,038.23  
6        4,642.83  
7        2,308.84  
8       31,625.95  
9      249,218.84  
10   1,744,318.34  
11   2,401,827.03  
12     298,764.86  
13       2,119.73  
14       6,025.41  
15      48,975.62  
16      12,032.65  
17   2,999,780.06  
18     223,303.98  
19   1,140,708.69  
20   1,044,695.01  
21     194,425.75  
22     490,850.96  
23   4,906,083.96  
24     218,267.39  
25     265,681.78  
26  16,248,523.35  
27     737,366.81

* website: 30 funds
* added 2 missing reports: 7012, 7011 (קרן ט, קרן י)
<br>
https://www.fnx.co.il/investors-relations-hebrew/investment-information-reports/reshimat-nichsey-hamevateah

In [129]:
url = "https://cdn.menoramivt.co.il/public/docs/20211205/520042540_b18012_p321.xlsx"
filename = "520042540_b18012_p321.xlsx"
ur.urlretrieve(url, filename)

URLError: <urlopen error [Errno 60] Operation timed out>

### הראל ביטוח - דוחות חסרים

In [113]:
pd.DataFrame(
    all_holdings_final[all_holdings_final["ParentCorpLegalId"] == '520004078'].groupby(["report_id", "ShortName", "ProductNum"])['שווי'].agg('sum').map('{:,.2f}'.format)
).sort_values("ProductNum", ascending=False).reset_index()

report_id                                    ShortName  ProductNum  \
0    2152323               הראל מסלול מחקה מדד תל אביב 35       13437   
1    2152329               הראל מסלול מחקה מדד תל בונד 60       13436   
2    2152324         הראל מסלול מחקה מדד ממשלתי צמוד 5-10       13435   
3    2152322                  הראל מסלול מחקה מדד S&P 500       13434   
4    2152321  הראל מסלול מחקה מדד ממשלתי שקלי ריבית קבועה       13433   
5    2152328                     הראל מסלול הוני ללא אג"ח       12981   
6    2152319    הראל מסלול אג"ח עד 20% במניות למקבלי קצבה       11404   
7    2152318                 הראל מסלול בסיסי למקבלי קצבה       11403   
8    2152315                   הראל מסלול לגילאי 60 ומעלה       11402   
9    2152317                 הראל מסלול לגילאי  50 עד 60        11401   
10   2152316                הראל ביטוח מסלול לגילאי עד 50       11400   
11   2152314   הראל מסלול אג"ח עד 20% במניות למקבלי קצבה         9566   
12   2152313                 הראל מסלול בסיסי למקבלי קצבה        9565   
13   2152320                     הראל מסלול לבני 60 ומעלה        9564   
14   2152310                     הראל מסלול לבני 50 עד 60        9563   
15   2152312                      הראל מסלול לבני 50 ומטה        9562   
16   2152311                    הראל מסלול פאסיבי -  כללי        9561   
17   2152309                         הראל מסלול אג"ח חו"ל        8592   
18   2152305               הראל-מסלול משתתף ברווחים מניות         197   
19   2152301       הראל-מסלול משתתף ברווחים כללי לא מניות         196   
20   2152306       הראל מסלול משתתף ברווחים שקלי טווח קצר         195   
21   2152307                      הראל מסלול אג"ח קונצרני         178   
22   2152297                 הראל מסלול אג"ח ממשלת ישראל          177   
23   2152303                              הראל מסלול חו"ל         163   
24   2152308                    הראל מסלול שיקלי טווח קצר          93   
25   2152300                            הראל מסלול מניות           92   
26   2152302                הראל מסלול אג"ח עד 10% במניות          91   
27   2152298                       הראל אג"ח עד 20% מניות          90   
28   2152299                   הראל מסלול אג"ח ללא מניות           89   
29   2152304                             הראל מסלול כללי           88   

             שווי  
0       25,604.19  
1       26,241.27  
2       24,556.88  
3      126,417.49  
4       12,385.38  
5      484,321.34  
6      366,509.58  
7      954,976.74  
8      241,868.35  
9      132,944.57  
10      54,252.97  
11     890,316.85  
12   1,595,212.12  
13   1,243,986.85  
14     791,703.78  
15   2,210,552.78  
16     145,663.99  
17      29,505.15  
18      86,915.55  
19      75,355.78  
20      44,374.82  
21     133,763.04  
22     281,196.10  
23     484,373.43  
24     513,488.17  
25   2,208,784.17  
26   1,717,240.36  
27   6,268,158.99  
28   1,392,743.61  
29  22,990,447.03

* website: 34 funds.<br><br>
<B> Manually added missing reports: 259010 259011 259012 259013 </B>
from website: https://www.harel-group.co.il/about/harel-group/harel/mesthtefet/Pages/list-of-assets.aspx

### כלל ביטוח - דוחות חסרים

In [114]:
pd.DataFrame(
    all_holdings_final[all_holdings_final["ParentCorpLegalId"] == '520024647'].groupby(["report_id", "ShortName", "ProductNum"])['שווי'].agg('sum').map('{:,.2f}'.format)
).sort_values("ProductNum", ascending=False).reset_index()

report_id                                          ShortName  ProductNum  \
0    2153455                          כלל ביטוח מחקה מדד S&P500       13346   
1    2153443                           כלל ביטוח שיקלי טווח קצר       13244   
2    2153450  כלל ביטוח למקבלי קצבה אג"ח עד 20% מניות - משתת...       12472   
3    2153441  כלל ביטוח פנסיונרים מקבלי קצבה אג"ח עד 20% מני...       12471   
4    2153442      כלל מסלול בסיסי למקבלי קצבה פוליסות לפני 2004       11350   
5    2153440         כלל מסלול לבני 60 ומעלה  פוליסות לפני 2004       11349   
6    2153444                  כלל לבני 50-60  פוליסות לפני 2004       11348   
7    2153438                 כלל לבני 50 ומטה פוליסות לפני 2004       11341   
8    2153454                        כלל מסלול בסיסי למקבלי קצבה        9767   
9    2153457                                     כלל - 60 ומעלה        9661   
10   2153437                                כלל - לבני 50 עד 60        9660   
11   2153453                                 כלל - לבני 50 ומטה        9659   
12   2153458                                       כלל - כהלכה          205   
13   2153456                                          כלל כללי           99   
14   2153449                                           כלל-שקלי          28   
15   2153448                                          כלל-מניות           9   
16   2153439                                         כלל-כללי 2           7   
17   2153436                              כלל אג"ח עד 15% מניות           6   
18   2153446                                          כלל-אג"ח            4   

             שווי  
0       60,196.59  
1       24,470.12  
2            0.47  
3            0.82  
4    2,197,872.01  
5      693,027.48  
6      386,921.93  
7      132,399.24  
8    1,326,029.64  
9    1,425,108.89  
10   1,193,400.89  
11   3,935,757.18  
12     188,267.48  
13  12,455,613.18  
14     483,783.29  
15   1,908,053.33  
16   1,280,599.57  
17  11,711,713.88  
18   2,645,195.44

* website: 21 funds
* missing reports: 14011, 14012
<br><br>
downloaded from their site: https://www.clalbit.co.il/aboutclalinsurance/generalreports/

## הלמן - פנסיה וגמל
?גמל - הבדל בסכום, האם כתוצאה מדוחות חסרים
<br>
₪16,136,063 בדוח מסכם
16,132,489.84 לפי סכום דוחות מסלול

הבדלים קטנים, לא מצאתי הסבר.
אולי הוציאו דוחות בשעות שונות של היום?

In [ ]:
16136063 - 16132489.84

In [ ]:
helman = all_holdings_final[(all_holdings_final["ParentCorpName"].str.startswith("הלמן")) &
                             (all_holdings_final["SystemName"] == "פנסיה")
                            ]
helman_by_report = pd.DataFrame(helman.groupby(["ProductNum", "ShortName"])["שווי"].agg('sum').map('{:,.2f}'.format)).reset_index()
helman_by_report

## מגדל פנסיה

In [91]:
migdal_p = all_holdings_final[(all_holdings_final["ParentCorpName"].str.startswith("מגדל")) &
                             (all_holdings_final["SystemName"] == "פנסיה")
                            ]
migdal_p_report = pd.DataFrame(migdal_p.groupby(["report_id", "ProductNum", "ShortName"])["שווי"].agg('sum')).reset_index()
migdal_p_report.merge(totals,
                     on="report_id",
                     how="left")

# migdal_p_report["שווי"].sum()

report_id  ProductNum                            ShortName          שווי  \
0    2148419       12147         מקפת אישית מניות למקבלי קצבה     17,477.05   
1    2148420        8803        מגדל מקפת אישית לבני 60 ומעלה  1,598,428.65   
2    2148421       12145               מקפת אישית למקבלי קצבה  4,079,662.67   
3    2148422        8602       מקפת אישית לזכאים קיימים לקצבה     78,442.11   
4    2148423       13572      מגדל מקפת אישית מחקה מדד S&P500    196,082.18   
5    2148424        8802        מגדל מקפת אישית לבני 50 עד 60  2,893,633.69   
6    2148425        2142                מגדל מקפת אישית מניות  2,234,914.47   
7    2148426        2207        מקפת אישית למקבלי קצבה קיימים  3,596,134.08   
8    2148427        2102                      מקפת אישית כללי 63,232,444.53   
9    2148428        9455            מקפת משלימה לבני 60 ומעלה     66,532.19   
10   2148429        2144                 מגדל מקפת אישית אג"ח    272,658.44   
11   2148430        9453             מקפת משלימה לבני 50 ומטה    369,795.19   
12   2148431        2149                     מקפת משלימה הלכה      6,197.51   
13   2148432        2146                    מקפת משלימה מניות     56,411.02   
14   2148433        2143        מגדל מקפת אישית שקלי טווח קצר    171,887.96   
15   2148434        2148                     מקפת משלימה אג"ח      5,483.95   
16   2148435        2112                 מגדל מקפת אישית הלכה  1,208,655.19   
17   2148436       12152              מקפת משלימה למקבלי קצבה     70,773.44   
18   2148437        9454            מקפת משלימה לבני 50 עד 60    115,934.18   
19   2148438        2145                     מקפת משלימה כללי    854,486.57   
20   2148439       13627     מגדל מקפת משלימה מחקה מדד S&P500      5,809.05   
21   2148440       12146          מקפת אישית הלכה למקבלי קצבה     25,543.52   
22   2148441       12148          מקפת אישית אג"ח למקבלי קצבה     11,549.68   
23   2148442        2147            מקפת משלימה שקלי טווח קצר      2,161.46   
24   2148443        2208       מקפת משלימה למקבלי קצבה קיימים    115,871.10   
25   2148444        8801         מגדל מקפת אישית לבני 50 ומטה 13,006,102.39   
26   2148445        8603  מקפת אישית הלכה למקבלי קצבה קיימים         505.68   

                            asset           sum  pct                pct_num  \
0   סה"כ סכום נכסי המסלול או הקרן     17,477.05    1                      1   
1   סה"כ סכום נכסי המסלול או הקרן  1,598,428.65    1                      1   
2   סה"כ סכום נכסי המסלול או הקרן  4,079,662.67    1                      1   
3   סה"כ סכום נכסי המסלול או הקרן     78,442.11    1                      1   
4   סה"כ סכום נכסי המסלול או הקרן    196,082.18    1                      1   
5   סה"כ סכום נכסי המסלול או הקרן  2,893,633.69    1                      1   
6   סה"כ סכום נכסי המסלול או הקרן  2,234,914.47    1                      1   
7   סה"כ סכום נכסי המסלול או הקרן  3,596,134.08    1                      1   
8   סה"כ סכום נכסי המסלול או הקרן 63,232,444.53 1.00     0.9999999999999999   
9   סה"כ סכום נכסי המסלול או הקרן     66,532.19    1                      1   
10  סה"כ סכום נכסי המסלול או הקרן    272,658.44    1                      1   
11  סה"כ סכום נכסי המסלול או הקרן    369,795.19    1                      1   
12  סה"כ סכום נכסי המסלול או הקרן      6,197.51 0.00  0.0013045484529668042   
13  סה"כ סכום נכסי המסלול או הקרן     56,411.02    1                      1   
14  סה"כ סכום נכסי המסלול או הקרן    171,887.96    1                      1   
15  סה"כ סכום נכסי המסלול או הקרן      5,483.95    1                      1   
16  סה"כ סכום נכסי המסלול או הקרן  1,208,655.19    1                      1   
17  סה"כ סכום נכסי המסלול או הקרן     70,773.53    1                      1   
18  סה"כ סכום נכסי המסלול או הקרן    115,934.18    1                      1   
19  סה"כ סכום נכסי המסלול או הקרן    854,486.57    1                      1   
20  סה"כ סכום נכסי המסלול או הקרן      5,809.05    1                      1   
21  סה"כ סכום נכסי המסלול או הקרן     25,543.52    1

* All pension funds from website are on file.<br>
<B> last Q there was also a discrepancy (2.1B, now 2.15B) </B>